src/output/location_insensitive.rs

```diff
@@ -0,0 +1,182 @@
+        .chain(all_facts.cfg_edge.iter().map(|&(_, q)| q))
+        .collect();
+
+    for &r in &all_facts.universal_region {
+        for &p in &all_points {
+            all_facts.region_live_at.push((r, p));
+        }
+    }
+
+    let result = Arc::new(Mutex::new(Output::new(dump_enabled)));
+
+    // Use a channel to send `all_facts` to one worker (and only one)
+    let (tx, rx) = mpsc::channel();
+    tx.send(all_facts).unwrap();
+    mem::drop(tx);
+    let rx = Mutex::new(rx);
+
+    timely::execute_from_args(vec![].into_iter(), {
+        let result = result.clone();
+        move |worker| {
+            // First come, first serve: one worker gets all the facts;
+            // the others get empty vectors.
+            let my_facts = rx.lock()
+                .unwrap()
+                .recv()
+                .unwrap_or_else(|_| AllFacts::default());
+
+            worker.dataflow::<(), _, _>(|scope| {
+                macro_rules! let_collections {
+                    (let ($($facts:ident,)*) = ..$base:expr;) => {
+                        let ($($facts),*) = (
+                            $(Collection::<_, _, isize>::new(
+                                $base.$facts
+                                    .to_stream(scope)
+                                    .map(|datum| (datum, Default::default(), 1)),
+                            ),)*
+                        );
+                    }
+                }
+
+                let_collections! {
+                    let (
+                        borrow_region,
+                        outlives,
+                        region_live_at,
+                    ) = ..my_facts;
+                }
+
+                // .decl subset(Ra, Rb) -- `R1 <= R2` holds
+                //
+                // subset(Ra, Rb) :- outlives(Ra, Rb, _P)
+                let subset = outlives
+                    .map(|(r_a, r_b, _p)| (r_a, r_b))
+                    .distinct_total();
+
+
+                // requires(R, L) :- borrow_region(R, L, _P).
+                let requires_base = borrow_region
+                    .map(|(r, l, _p)| (r, l))
+                    .distinct_total();
+
+                // requires(R2, L) :- requires(R1, L), subset(R1, R2).
+                let requires = requires_base.iterate(|requires| {
+                    let subset = subset.enter(&requires.scope());
+                    let requires_base = requires_base.enter(&requires.scope());
+
+                    let requires1 = requires_base.clone();
+
+                    let requires2 = requires
+                        .join(&subset)
+                        .map(|(_r1, l, r2)| (r2, l));
+
+                    requires1.concat(&requires2).distinct_total()
+                });
+
+                // borrow_live_at(L, P) :-
+                //     requires(R, L), region_live_at(R, P)
+                let borrow_live_at = requires
+                    .join(&region_live_at)
+                    .map(|(_r, l, p)| (l, p));
```

Comment:
you're missing a `distinct_total` here I believe

Reply:
You're right, probably a copy and paste from the previous part of the code where distinct_total is done in the concat line. Fixed!!!